src/panel/direction/DirectionForm.jsx

```diff
@@ -30,10 +30,12 @@ export default class DirectionForm extends React.Component {
     if (isMobileDevice() || this.props.isInitializing) {
       return;
     }
+
     const { origin, destination, originInputText, destinationInputText } = this.props;
+
     if (!originInputText && (prevProps.destination !== destination || prevProps.isInitializing)) {
       this.originRef.current.focus();
-    } else if (!destinationInputText && prevProps.origin !== origin) {
+    } else if (!destinationInputText && prevProps.origin !== origin && origin) {
       this.destinationRef.current.focus();
     }
```

Comment:
I have a question, probably unrelated.
  
In these 2 conditions, it's unclear for me why we would need to check for `!originInputText` and `!destinationInputText`, instead of `!origin` and `!destination`. Especially when the input text may be populated asynchronously for latlon objects that require an API call to fetch the address.  

Am I missing something ? And maybe some comments would help to make this clearer.

Reply:
I agree, I remove destinationInputText and originInputText comparisons :+1: